### PR TITLE
Np 3603 post project using unit id pr

### PR DESCRIPTION
**Pull request by axthosarouris**

### CODE REVIEW

File: cristin-project/src/main/java/no/unit/nva/cristin/projects/model/nva/NvaContributor.java [hunk 1]
```diff
@@ -51,12 +68,82 @@ public boolean equals(Object o) {
...
-                && getIdentity().equals(that.getIdentity())
-                && getAffiliation().equals(that.getAffiliation());
+               && getIdentity().equals(that.getIdentity())
+               && getAffiliation().equals(that.getAffiliation());
     }
 
-    @Override
-    public int hashCode() {
-        return Objects.hash(getType(), getIdentity(), getAffiliation());
+    public CristinPerson toCristinPerson() {
```
Comment (torarnet) [inline cristin-project/src/main/java/no/unit/nva/cristin/projects/model/nva/NvaContributor.java:75]: Identity is a Person object. Maybe we can create a Person.toCristinPerson? We can extract roles separate as that is not part of identity

File: cristin-project/src/main/java/no/unit/nva/cristin/projects/model/nva/NvaContributor.java [hunk 2]
```diff
@@ -51,12 +68,82 @@ public boolean equals(Object o) {
...
+        return contributorTypeCanBeMappedToCristinRole()
+                   ? getCristinRoles()
+                   : Collections.emptyList();
+    }
+
+    private boolean contributorTypeCanBeMappedToCristinRole() {
+        return getCristinRole(getType()).isPresent();
+    }
+
+    private List<CristinRole> extractCristinRoles(NvaContributor contributor) {
```
Comment (torarnet) [inline cristin-project/src/main/java/no/unit/nva/cristin/projects/model/nva/NvaContributor.java:95]: Duplicate?

File: cristin-project/src/main/java/no/unit/nva/cristin/projects/model/nva/NvaContributor.java [hunk 3]
```diff
@@ -51,12 +68,82 @@ public boolean equals(Object o) {
...
+        return cristinUnit;
+    }
+
+    private Optional<String> extractUnitIdentifier() {
+        return Optional.of(getAffiliation())
+            .map(Organization::getId).map(UriUtils::extractLastPathElement)
+            .filter(identifier -> UNIT_ID_PATTERN.matcher(identifier).matches());
+    }
+
+    private CristinInstitution convertOrganizationToCristinInstitution(Organization organization) {
```
Comment (torarnet) [inline cristin-project/src/main/java/no/unit/nva/cristin/projects/model/nva/NvaContributor.java:142]: Can be a Organization.toCristinInstitution?

File: cristin-project/src/main/java/no/unit/nva/cristin/projects/model/nva/NvaContributor.java [hunk 4]
```diff
@@ -51,12 +68,82 @@ public boolean equals(Object o) {
...
+        return extractUnitIdentifier().isPresent();
+    }
+
+    private CristinUnit toCristinUnit(String unitIdentifier) {
+        CristinUnit cristinUnit = new CristinUnit();
+        cristinUnit.setCristinUnitId(unitIdentifier);
+        return cristinUnit;
+    }
+
+    private Optional<String> extractUnitIdentifier() {
```
Comment (torarnet) [inline cristin-project/src/main/java/no/unit/nva/cristin/projects/model/nva/NvaContributor.java:136]: method in Organization? Organization identifier is of type department

**Review by torarnet — COMMENTED** [review]
Have some comments i want you to give some feedback on. If it seems right I can merge this and do those changes myself

File: cristin-project/src/main/java/no/unit/nva/cristin/projects/model/nva/NvaContributor.java (hunk 4 above)
Comment (axthosarouris) [reply cristin-project/src/main/java/no/unit/nva/cristin/projects/model/nva/NvaContributor.java:134]: see previous comment

**Review by torarnet — APPROVED** [review]

**Pull request merged**